### PR TITLE
Extract SQL logic from WebserviceRequest to unit test it

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Extract SQL logic from WebserviceRequest to unit test it
| Type?         | improvement
| Category?     | WS
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | 
| How to test?  | Travis is green

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19389)

### CODE REVIEW

File: tests-legacy/Unit/Classes/WebserviceSQLUtilsTest.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public function testGetSimpleFilterStatement1()
+    {
+        $a = \SQLUtils::getSQLRetrieveFilter('name', 'a', 'main.');
+        $this->assertEquals(' AND `main`.`name` = "a"' . PHP_EOL, $a);
+    }
+
+    public function testGetSimpleFilterStatement2()
+    {
+        $a = \SQLUtils::getSQLRetrieveFilter('price', '18.2', 'main.');
+        $this->assertEquals(' AND `main`.`price` LIKE "18.2"' . PHP_EOL, $a);
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Classes/WebserviceSQLUtilsTest.php]: I don't understand why LIKE is used here :thinking: 
Is it related to a pull request where someone change the ternary order?

Comment (matks) [reply tests-legacy/Unit/Classes/WebserviceSQLUtilsTest.php]: Yes, this test aims to validate PR https://github.com/PrestaShop/PrestaShop/pull/19249

And this LIKE is exactly what we want to remove 😄

File: tests-legacy/Unit/Classes/WebserviceSQLUtilsTest.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+
+    public function testGetSimpleFilterStatement3()
+    {
+        $a = \SQLUtils::getSQLRetrieveFilter('name', '[19.2, 19.8]', 'main.');
+        $this->assertEquals(' AND `main`.`name` BETWEEN "19.2" AND " 19.8"' . PHP_EOL, $a);
```
Comment (PierreRambaud) [inline tests-legacy/Unit/Classes/WebserviceSQLUtilsTest.php]: Can you test all the possibilities of this function? (Or I can update your pull request if you want :) )

Comment (matks) [reply tests-legacy/Unit/Classes/WebserviceSQLUtilsTest.php]: Update 😄feel like home !

File: classes/webservice/SQLUtils.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
```
Comment (Progi1984) [inline classes/webservice/SQLUtils.php]:
```suggestion
 */
 declare(strict_types=1);
 
```

**Review by PierreRambaud — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Comment by Progi1984** [comment]
@matks PHPCSFixer is not happy ;)

**Comment by matks** [comment]
> @matks PHPCSFixer is not happy ;)

Grrrrr 😡

**Comment by Progi1984** [comment]
Ping @matks

**Review by Progi1984 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @matks

**Pull request merged**